pkg/operation/botanist/controlplane.go

```diff
@@ -201,7 +213,9 @@ func (b *Botanist) DeleteClusterAutoscaler(ctx context.Context) error {
 // * kube-apiserver
 // * kube-controller-manager
 func (b *Botanist) WakeUpControlPlane(ctx context.Context) error {
-	client := b.K8sSeedClient.Client()
+	// use direct client here, as cached client sometimes causes scale functions not to work properly
```

Comment:
Hm, interesting. Is there an issue in the controller-runtime that we can link here? If not, could you open one?

Reply:
Hmm, actually I think, this is more a problem on our side in the `kubernetes.Scale*` funcs. But I didn't investigate further, as I thought, using the direct client for such calls, that are only executed once/twice per day per Shoot should not harm too much.
What we could try, is rather use `client.Patch` for setting the replica count instead of `client.Update` as long as there is no support for the `scale` subresource in controller-runtime (see https://github.com/kubernetes-sigs/controller-runtime/issues/172), but I'm not actually sure, that this would help.

Reply:
Hm, okay, thanks. Let's keep it like this until the subresources are supported in the controller-runtime. Resolving this comment.